RecoLocalTracker/SiPhase2VectorHitBuilder/src/VectorHitBuilderAlgorithm.cc

```diff
@@ -0,0 +1,553 @@
+      for (const auto& vh : vhsInStack_AccRej) {
+        if (vh.second == true) {
+          vhsInStack_Acc.push_back(vh.first);
+          std::push_heap(vhsInStack_Acc.begin(), vhsInStack_Acc.end());
+          totalAccepted += 1;
+        } else if (vh.second == false) {
+          vhsInStack_Rej.push_back(vh.first);
+          totalRejected += 1;
+        }
+      }
+
+      //ERICA:: to be checked with map!
+      //sorting vhs for best chi2
+      std::sort_heap(vhsInStack_Acc.begin(), vhsInStack_Acc.end());
+
+      tempVHAcc[detIdStack] = vhsInStack_Acc;
+      tempVHRej[detIdStack] = vhsInStack_Rej;
+#ifdef EDM_ML_DEBUG
+      LogTrace("VectorHitBuilderAlgorithm")
+          << "For detId #" << detIdStack.rawId() << " the following VHits have been accepted:";
+      for (const auto& vhIt : vhsInStack_Acc) {
+        LogTrace("VectorHitBuilderAlgorithm") << "accepted VH: " << vhIt;
+      }
+      LogTrace("VectorHitBuilderAlgorithm")
+          << "For detId #" << detIdStack.rawId() << " the following VHits have been rejected:";
+      for (const auto& vhIt : vhsInStack_Rej) {
+        LogTrace("VectorHitBuilderAlgorithm") << "rejected VH: " << vhIt;
+      }
+#endif
+    }
+  }
+  loadDetSetVector(tempVHAcc, vhAcc, totalAccepted);
+  loadDetSetVector(tempVHRej, vhRej, totalRejected);
+
+  LogDebug("VectorHitBuilderAlgorithm") << "End run VectorHitBuilderAlgorithm ... \n";
+}
+
+bool VectorHitBuilderAlgorithm::checkClustersCompatibilityBeforeBuilding(
+    edm::Handle<edmNew::DetSetVector<Phase2TrackerCluster1D>> clusters,
+    const detset& theLowerDetSet,
+    const detset& theUpperDetSet) const {
+  if (theLowerDetSet.size() == 1 && theUpperDetSet.size() == 1)
+    return true;
+
+  //order lower clusters in u
+  std::vector<Phase2TrackerCluster1D> lowerClusters;
+  lowerClusters.reserve(theLowerDetSet.size());
+  if (theLowerDetSet.size() > 1)
+    LogDebug("VectorHitBuilderAlgorithm") << " more than 1 lower cluster! " << std::endl;
+  if (theUpperDetSet.size() > 1)
+    LogDebug("VectorHitBuilderAlgorithm") << " more than 1 upper cluster! " << std::endl;
+  for (const_iterator cil = theLowerDetSet.begin(); cil != theLowerDetSet.end(); ++cil) {
+    Phase2TrackerCluster1DRef clusterLower = edmNew::makeRefTo(clusters, cil);
+    lowerClusters.push_back(*clusterLower);
+  }
+  return true;
+}
+
+bool VectorHitBuilderAlgorithm::checkClustersCompatibility(Local3DPoint& poslower,
+                                                           Local3DPoint& posupper,
+                                                           LocalError& errlower,
+                                                           LocalError& errupper) const {
+  return true;
+}
+
+//----------------------------------------------------------------------------
+//ERICA::in the DT code the global position is used to compute the alpha angle and put a cut on that.
+std::vector<std::pair<VectorHit, bool>> VectorHitBuilderAlgorithm::buildVectorHits(
```

Comment:
why not return a pair of vectors?
It looks like the only purpose of a vector of pairs is to expand it to a pair of vectors

Reply:
The function is now return type `void` as the VH collections are filled directly in it